tests/suites/test_suite_pk.pk_can_do_psa.data

```diff
@@ -0,0 +1,260 @@
+depends_on:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:PSA_HAVE_ALG_SOME_ECDSA:!MBEDTLS_ECP_DP_SECP224K1_ENABLED
+pk_can_do_psa:MBEDTLS_PK_ECKEY:PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):PSA_KEY_USAGE_EXPORT:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_SHA_256):PSA_KEY_USAGE_SIGN_HASH:1
+
+PK can do PSA: MBEDTLS_PK_ECKEY (no priv), check sign hash - Fail
+depends_on:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:PSA_HAVE_ALG_SOME_ECDSA:!MBEDTLS_ECP_DP_SECP224K1_ENABLED
+pk_can_do_psa:MBEDTLS_PK_ECKEY:PSA_KEY_TYPE_ECC_PUBLIC_KEY(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):PSA_KEY_USAGE_EXPORT:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_SHA_256):PSA_KEY_USAGE_SIGN_HASH:0
+
+PK can do PSA: MBEDTLS_PK_ECKEY, check sign hash, wrong alg - Fail
+depends_on:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:PSA_HAVE_ALG_SOME_ECDSA:!MBEDTLS_ECP_DP_SECP224K1_ENABLED
+pk_can_do_psa:MBEDTLS_PK_ECKEY:PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):PSA_KEY_USAGE_EXPORT:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS:PSA_ALG_ECDH:PSA_KEY_USAGE_SIGN_HASH:0
+
+PK can do PSA: MBEDTLS_PK_ECKEY, check verify hash - Success
+depends_on:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:PSA_HAVE_ALG_SOME_ECDSA:!MBEDTLS_ECP_DP_SECP224K1_ENABLED
+pk_can_do_psa:MBEDTLS_PK_ECKEY:PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):PSA_KEY_USAGE_EXPORT:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_SHA_256):PSA_KEY_USAGE_VERIFY_HASH:1
+
+PK can do PSA: MBEDTLS_PK_ECKEY, check verify hash - Fail
+depends_on:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:PSA_HAVE_ALG_SOME_ECDSA:!MBEDTLS_ECP_DP_SECP224K1_ENABLED
+pk_can_do_psa:MBEDTLS_PK_ECKEY:PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):PSA_KEY_USAGE_EXPORT:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS:PSA_ALG_ECDH:PSA_KEY_USAGE_VERIFY_HASH:0
+
+PK can do PSA: MBEDTLS_PK_ECKEY, check encrypt - Fail
+depends_on:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:PSA_HAVE_ALG_SOME_ECDSA:!MBEDTLS_ECP_DP_SECP224K1_ENABLED
+pk_can_do_psa:MBEDTLS_PK_ECKEY:PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):PSA_KEY_USAGE_EXPORT:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS:PSA_ALG_NONE:PSA_KEY_USAGE_ENCRYPT:0
+
+PK can do PSA: MBEDTLS_PK_ECKEY, check decrypt - Fail
+depends_on:PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY:PSA_HAVE_ALG_SOME_ECDSA:!MBEDTLS_ECP_DP_SECP224K1_ENABLED
+pk_can_do_psa:MBEDTLS_PK_ECKEY:PSA_KEY_TYPE_ECC_KEY_PAIR(MBEDTLS_TEST_PSA_ECC_ONE_FAMILY):PSA_KEY_USAGE_EXPORT:MBEDTLS_PK_ALG_ECDSA(PSA_ALG_ANY_HASH):PSA_ALG_NONE:MBEDTLS_TEST_PSA_ECC_ONE_CURVE_BITS:PSA_ALG_NONE:PSA_KEY_USAGE_DECRYPT:0
+
+# Legacy contexts have private keys which do not support derivation.
+PK can do PSA: MBEDTLS_PK_ECKEY, check derive - Fail
```

Comment:
A legacy ECC context can do derivation. Not through PK, because PK doesn't have a function for it, but it can be done by importing the key into PSA. So this test case should be a success.

On a related note, please add a test case with a public-only key, for which DERIVE with ECDH should fail.

On a related note, `DERIVE_PUBLIC` test cases are missing. This should succeed with ECDH for an ECC key pair context and for an ECC public-only context, and fail for other key types.